### ISSUE TITLE
[Feature request] Hide and Unhide PowerPoint Slides

### ISSUE THREAD

**Issue by laresbernardo** [davidgohel/officer#622]
I've reviewed all the available documentation and haven't seen an option to hide or unhide a slide. This is a valuable functionality and I wouldn't expect it to be too complex to include in the package.

**Use Case:** I've included an appendix slide that contains the code to replicate the deck. There's no need to include this slide in a presentation but it's super useful to have in the deck.

Thanks for this amazing package!

**Comment by davidgohel** [comment]
> I wouldn't expect it to be too complex

How do you estimate that ? Do you have pointers that show how to implement that ?

**Comment by laresbernardo** [comment]
You are right, I don't. Sorry if that sounded rude. I extrapolated from the advanced functionalities I've seen already implemented in officer and thought it was a matter of finding the right "command" to hide and unhide slides. No foundations to back the comment. Hope it's not complicated though!

**Comment by markheckmann** [comment]
#### Technical 

On a technical level, it appears that a slide is hidden (shown) by setting the `show` attribute in the `p:sld` node to `0` (`1`) (in the slide xml file). 

```xml
<p:sld xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main" 
xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships" 
xmlns:p="http://schemas.openxmlformats.org/presentationml/2006/main" show="0">
...
</p:sld>
```

----

#### API 

Does someone have a suggestion for the API? Some options could be:

- a `visible` arg in `add_slide()` which defaults to `TRUE`
- a new function `slide_visible()` to get and set a slide's visibility (e.g. `hide` and `show` arg taking slide indexes)

_Examples_:

```r

add_slide(x, "Title Slide", visible = FALSE)

slide_visible(x)
[1] TRUE TRUE TRUE TRUE

slide_visible(x, hide = 2:3)
slide_visible(x, show = c(1,4))

slide_visible(x)[2:3] <- FALSE
```
----
**@davidgohel Please feel free to assign this issue to me**

**Comment by laresbernardo** [comment]
@markheckmann of course, as I shared in my first comment, I'm generating a slide at the end of the deck that contains the code to replicate the deck. It also contains a full JSON formatted string with all the results (not shown in the nice slides but that may be useful). Given these decks I'm working on are ready to be pitched/shared, I wouldn't want them to be shown while presenting.

**Comment by davidgohel** [comment]
@laresbernardo in this use case, using comments/notes instead of hidden slide seems more relevant to me but it's only a point of view.

You can have a look at `set_notes()`

**Comment by markheckmann** [comment]
@laresbernardo Can you please close this issue? I do not have the right to do so. 

Was completed via Closed via #621

**Comment by laresbernardo** [comment]
Thanks a lot! Closing ticket. I have already tested it on my end and it is working as expected.
